graphql/src/main/java/io/micronaut/configuration/graphql/ws/GraphQLWsRequest.java

```diff
@@ -15,24 +15,24 @@
  */
 package io.micronaut.configuration.graphql.ws;
 
-import io.micronaut.configuration.graphql.GraphQLRequestBody;
 import io.micronaut.core.annotation.Nullable;
 
 /**
  * Class to handle the message to and from the websocket.
  *
+ * @param <P> The payload type
  * @author Gerard Klijs
  * @since 1.3
  */
-public class GraphQLWsRequest {
+public abstract class GraphQLWsRequest<P> {
```

Comment:
this is a breaking change. We cannot add this until the next mayor version.

Reply:
I know, that's why I planned this to be in 4.0.

Reply:
yes, please.